extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js

```diff
@@ -85,13 +97,19 @@ export class AmpRecaptchaService {
   /**
    * Function to register as a dependant of the AmpRecaptcha serivce.
    * Used to create/destroy recaptcha boostrap iframe.
-   * @param {!Element} element
+   * @param {string} sitekey
    * @return {Promise}
    */
-  register(element) {
+  register(sitekey) {
     this.registeredElementCount_++;
     if (!this.iframeLoadPromise_) {
-      this.initialize_(element);
+      return new Promise((resolve, reject) => {
+        this.initialize_(sitekey).then(() => {
+          this.iframeLoadPromise_.then(() => {
+            resolve();
+          }).catch(err => reject(err));
```

Comment:
https://github.com/ampproject/amphtml/blob/493ea5fd3915b9e5efd9b070bf71d0dc3d48e424/src/custom-element.js#L1093

customElement handles promise error. I don't believe there's need for you to catch the error and call promise.reject here. 

One thing we need to think about though: is that do we want to throw the same error multiple times if there are multiple `<amp-recaptcha-input>` on the page. Which I think we should.

Reply:
> customElement handles promise error. I don't believe there's need for you to catch the error and call promise.reject here.

I updated this implementation to follow a syntax without catching, and simply returning the `iframeLoadPromise_` in the `.then()` block. 😄 

> One thing we need to think about though: is that do we want to throw the same error multiple times if there are multiple <amp-recaptcha-input> on the page. Which I think we should.

With the new syntax I used, if the `iframeLoadPromise_` promise fails, then every component that registers will simply return the fails promise, and then the catch block will be executed for that component. Thus doing what you think should happen (and I agree with) 👍